### PR TITLE
Modify iterators to work across shards

### PR DESCRIPTION
**Pull request by jsternberg**
Aux iterators now ask the iterator creator what series will be returned
and determine which aux fields to create based on the results.

The `tsdb.Shards` struct also creates a call iterator around the
iterators returned from each shard.

### CODE REVIEW

File: influxql/iterator.gen.go [hunk 1]
```diff
@@ -418,6 +418,7 @@ type floatFillIterator struct {
 	curTime    int64
 	startTime  int64
 	endTime    int64
+	auxFields  []interface{}
```
Comment (jwilder) [inline influxql/iterator.gen.go:421]: What are `auxFields`?  Would be useful to have a comment w/ some examples.

Comment (benbjohnson) [inline influxql/iterator.gen.go:421]: Auxiliary fields are fields attached to a point. For example in the query:

```
SELECT min(value), fieldA, fieldB FROM cpu
```

The point represents `min(value)` but the auxiliary fields are `fieldA` and `fieldB` which are the values at that particular data point.

There's some write up of the query engine in the godocs but I'll write up a more substantial overview in an `influxql` README or godoc.

**Comment by benbjohnson** [comment]
:+1:

File: influxql/iterator.gen.go (hunk 1 above)
Comment (jsternberg) [inline influxql/iterator.gen.go:421]: This is mostly there as the fill iterator needs to send nil values for the auxiliary fields and I don't think it's good to allocate a new array with the same values for every point, so I store the nil auxiliary values for the series here so the same ones can be sent with every point that gets created.

Comment (jwilder) [inline influxql/iterator.gen.go:421]: I see.  Got it.

**Pull request merged**